metrics-operator/controllers/common/analysis/taget_evaluator_test.go

```diff
@@ -0,0 +1,146 @@
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestTargetEvaluator_Evaluate(t *testing.T) {
+	compValue15 := resource.NewQuantity(15, resource.DecimalSI)
+	compValue20 := resource.NewQuantity(20, resource.DecimalSI)
+	tests := []struct {
+		name string
+		val  float64
+		t    v1alpha3.Target
+		want types.TargetResult
+	}{
+		{
+			name: "failure scenario",
+			val:  10.0,
+			t: v1alpha3.Target{
+				Failure: &v1alpha3.Operator{
+					LessThan: &v1alpha3.OperatorValue{
+						FixedValue: *compValue15,
+					},
+				},
+			},
+			want: types.TargetResult{
+				FailureResult: types.OperatorResult{
+					Operator: v1alpha3.Operator{
+						LessThan: &v1alpha3.OperatorValue{
+							FixedValue: *compValue15,
+						},
+					},
+					Fulfilled: true,
+				},
+				Warning: false,
+				Pass:    false,
+			},
+		},
+		{
+			name: "warning scenario",
+			val:  17.0,
+			t: v1alpha3.Target{
+				Failure: &v1alpha3.Operator{
+					LessThan: &v1alpha3.OperatorValue{
+						FixedValue: *compValue15,
+					},
+				},
+				Warning: &v1alpha3.Operator{
+					LessThan: &v1alpha3.OperatorValue{
+						FixedValue: *compValue20,
+					},
+				},
+			},
+			want: types.TargetResult{
+				FailureResult: types.OperatorResult{
+					Operator: v1alpha3.Operator{
+						LessThan: &v1alpha3.OperatorValue{
+							FixedValue: *compValue15,
+						},
+					},
+					Fulfilled: false,
+				},
+				WarningResult: types.OperatorResult{
+					Operator: v1alpha3.Operator{
+						LessThan: &v1alpha3.OperatorValue{
+							FixedValue: *compValue20,
+						},
+					},
+					Fulfilled: true,
+				},
+				Warning: true,
+				Pass:    false,
+			},
+		},
+		{
+			name: "pass scenario",
+			val:  27.0,
+			t: v1alpha3.Target{
+				Failure: &v1alpha3.Operator{
+					LessThan: &v1alpha3.OperatorValue{
+						FixedValue: *compValue15,
+					},
+				},
+				Warning: &v1alpha3.Operator{
+					LessThan: &v1alpha3.OperatorValue{
+						FixedValue: *compValue20,
+					},
+				},
+			},
+			want: types.TargetResult{
+				FailureResult: types.OperatorResult{
+					Operator: v1alpha3.Operator{
+						LessThan: &v1alpha3.OperatorValue{
+							FixedValue: *compValue15,
+						},
+					},
+					Fulfilled: false,
+				},
+				WarningResult: types.OperatorResult{
+					Operator: v1alpha3.Operator{
+						LessThan: &v1alpha3.OperatorValue{
+							FixedValue: *compValue20,
+						},
+					},
+					Fulfilled: false,
+				},
+				Warning: false,
+				Pass:    true,
+			},
+		},
+		{
+			name: "pass scenario - only failed defined",
+			val:  17.0,
+			t: v1alpha3.Target{
+				Failure: &v1alpha3.Operator{
+					LessThan: &v1alpha3.OperatorValue{
+						FixedValue: *compValue15,
+					},
+				},
+			},
+			want: types.TargetResult{
+				FailureResult: types.OperatorResult{
+					Operator: v1alpha3.Operator{
+						LessThan: &v1alpha3.OperatorValue{
+							FixedValue: *compValue15,
+						},
+					},
+					Fulfilled: false,
+				},
+				Warning: false,
+				Pass:    true,
+			},
+		},
+	}
```

Comment:
please also add a test case where neither `failure` or `warning` is set

Reply:
you mean in the Target in AnalysisDefinition?

Reply:
yes please